src/tscache.ts

```diff
@@ -264,7 +264,7 @@ export class TsCache
 		if (this.ambientTypesDirty)
 			this.context.info(yellow("ambient types changed, redoing all semantic diagnostics"));
 
-		typeNames.forEach(this.typesCache.touch, this);
+		_.each(typeNames, (name) => this.typesCache.touch(name));
```

Comment:
This line was responsible I think. I reverted all forEach changes just in case
![image](https://user-images.githubusercontent.com/1617834/171038482-9de01deb-9a3f-421a-ab03-705402292eb4.png)

Reply:
the bug here might be that I didn't put the correct `this` context here when I re-wrote it -- let me double-check that shortly

Reply:
yep the problem is `this`. similar to the `Array#map` statements in `index.ts`, it should be:
```suggestion
		typeNames.forEach(this.typesCache.touch, this.typesCache);
```

Can confirm that works without issues